### PR TITLE
Add libphonenumber-js

### PR DESCRIPTION
**Pull request by halt-hammerzeit**
Pull request for issue: #9962

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/halt-hammerzeit/libphonenumber-js
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 10
   - Count of watchers: 2
   - Count of forks: 0

# Essential checklist
 * [x] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [ ] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [ ] The parent of the commit(s) in the PR is not old than 3 days.
 * [ ] Pull request is sending from a non-master branch with meaningful name.
 * [ ] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/libphonenumber-js/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    "iphone",
+    "international",
+    "libphonenumber"
+  ],
+  "npmName": "libphonenumber-js",
+  "npmFileMap": [
+    {
+      "basePath": "bundle",
+      "files": [
+        "libphonenumber-js.min.js"
```
Comment (x09326) [inline ajax/libs/libphonenumber-js/package.json]: @halt-hammerzeit 
I think use `libphonenumber-js*` is better.

**Comment by halt-hammerzeit** [comment]
@x09326 changed 	`libphonenumber-js.min.js` -> ibphonenumber-js*

File: ajax/libs/libphonenumber-js/package.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "libphonenumber-js",
+  "description": "A simpler (and smaller) rewrite of Google Android's popular libphonenumber library",
+  "version": "0.2.3",
+  "author": {
+    "name": "Halt Hammerzeit",
+    "email": "halt.hammerzeit.at@gmail.com"
+  },
+  "filename": "libphonenumber-js.js",
```
Comment (nolsherry) [inline ajax/libs/libphonenumber-js/package.json]: Please modified `libphonenumber-.js` to `libphonenumber-js.min.js` in `filename` field, thanks!

File: ajax/libs/libphonenumber-js/package.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  "homepage": "https://github.com/halt-hammerzeit/libphonenumber-js",
```
Comment (nolsherry) [inline ajax/libs/libphonenumber-js/package.json]: The homepage url is the same as github repository url.
I think that the homepage url `https://halt-hammerzeit.github.io/libphonenumber-js/` is better.

**Comment by halt-hammerzeit** [comment]
@nolsherry Thanks for the tips, I updated the pull request with the two fixes you mentioned.

**Comment by halt-hammerzeit** [comment]
@x09326 Yeah, I could, but this would require me downloading this repo on my machine.
Don't you have "squash and merge" option when clicking the triangle on the merge button?
https://github.com/blog/2141-squash-your-commits
If not then I can recreate this pull request with a single commit (via github GUI).

**Comment by x09326** [comment]
@PeterDaveHello Can we just use squash and merge in this case?

**Comment by PeterDaveHello** [comment]
@x09326 What kind of case do you mean?

**Comment by x09326** [comment]
@PeterDaveHello 
Because @halt-hammerzeit said above that if we don't use squash and merge then he need to download the repo to his machine

**Comment by PeterDaveHello** [comment]
Yes, of course, we can squash for @halt-hammerzeit, just make sure the other parts are good!

**Comment by PeterDaveHello** [comment]
@kennynaoh

**Comment by kennynaoh** [comment]
Hello @halt-hammerzeit Could you remove the `version` field in package.json or upload the library to let the test pass? Thanks!

**Comment by halt-hammerzeit** [comment]
@kennynaoh Hello.
Of course, removed the `version` field.

**Comment by PeterDaveHello** [comment]
@x09326 please review and help @halt-hammerzeit rebase his branch with the latest master branch, and also squash the commits (as you already need to fetch and rebase the branch), thanks.

**Comment by halt-hammerzeit** [comment]
@PeterDaveHello I can recreate this pull request from scratch if needed: it only has a single `package.json` file.

**Comment by PeterDaveHello** [comment]
@halt-hammerzeit cool, let's do it, please, thanks :)

**Comment by PeterDaveHello** [comment]
btw, please send pull request from a non-master branch, thanks :)

**Comment by halt-hammerzeit** [comment]
@PeterDaveHello 
> btw, please send pull request from a non-master branch, thanks :)

I don't understand.
Why isn't `master` branch suitable.

**Comment by halt-hammerzeit** [comment]
In the meantime, I created the new PR https://github.com/cdnjs/cdnjs/pull/10200, and I'm closing this one

**Comment by PeterDaveHello** [comment]
@halt-hammerzeit 

1. It'll make you very hard to send any further pull request again, since the diverse of "Your master branch" and "CDNJS's master branch", you'll spend more time on fixing that.

2. meaningful name of a branch help build cleaner and clearer commit history.

3. It's one of the most common convention of various git workflows that you should keep branch from upstream just the same as what it is as upstream, unless, you are going to build your own version/release, or you should create a new feature branch.

Ref:
https://help.github.com/articles/about-pull-requests/
https://help.github.com/articles/creating-a-pull-request/

**Comment by halt-hammerzeit** [comment]
@PeterDaveHello Fair point, thanks for the explanation.
I redid the PR: https://github.com/cdnjs/cdnjs/pull/10201

Must be a lot of work for your team to spend such amounts of time for each library being added.

My workflow was (you might publish this as a guide for simple projects):
* Fork the repo
* Create the branch using the "Branches" dropdown
* Go to `ajax/libs`
* Click `Create file`
* Enter `libphonenumber-js/package.json`
* Post `package.json`
* Commit and create a Pull Request

**Comment by PeterDaveHello** [comment]
@halt-hammerzeit thanks! I wonder if you would like to send a pull request about the guide?

Actually we already have some:
 - https://github.com/cdnjs/cdnjs#adding-a-new-or-updating-an-existing-library
 - https://github.com/cdnjs/cdnjs/blob/master/.github/CONTRIBUTING.md

One of the problem is I'm not a native speaker, and my English skill is not good enough, so the doc is not completed yet ...

**Comment by halt-hammerzeit** [comment]
@PeterDaveHello I wrote down a simple guide. See here: https://github.com/cdnjs/cdnjs/pull/10206
If you find it appropriate you can edit it or merge it.

**Comment by PeterDaveHello** [comment]
Thanks @halt-hammerzeit

**Pull request closed without merging**